addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -169,33 +169,54 @@ def _get_seen_list_sms(self):
         self.ensure_one()
         target = self.env[self.mailing_model_real]
 
+        partner_field = False
         if issubclass(type(target), self.pool['mail.thread.phone']):
             phone_fields = ['phone_sanitized']
         elif issubclass(type(target), self.pool['mail.thread']):
             phone_fields = target._sms_get_number_fields()
+            partner_field = target._sms_get_partner_fields()
         else:
             phone_fields = []
             if 'mobile' in target._fields:
                 phone_fields.append('mobile')
             if 'phone' in target._fields:
                 phone_fields.append('phone')
-        if not phone_fields:
+        if not phone_fields and not partner_field:
             raise UserError(_("Unsupported %s for mass SMS", self.mailing_model_id.name))
 
-        query = """
-            SELECT %(select_query)s
-              FROM mailing_trace trace
-              JOIN %(target_table)s target ON (trace.res_id = target.id)
-             WHERE (%(where_query)s)
-             AND trace.mass_mailing_id = %%(mailing_id)s
-             AND trace.model = %%(target_model)s
-        """
-        query = query % {
-            'select_query': 'target.id, ' + ', '.join('target.%s' % fname for fname in phone_fields),
-            'where_query': ' OR '.join('target.%s IS NOT NULL' % fname for fname in phone_fields),
-            'target_table': target._table
-        }
-        params = {'mailing_id': self.id, 'target_model': self.mailing_model_real}
+        if phone_fields:
+            query = """
+                SELECT %(select_query)s
+                  FROM mailing_trace trace
+                  JOIN %(target_table)s target ON (trace.res_id = target.id)
+                 WHERE (%(where_query)s)
+                   AND trace.mass_mailing_id = %%(mailing_id)s
+                   AND trace.model = %%(target_model)s
+            """
+            query = query % {
+                'select_query': 'target.id, ' + ', '.join('target.%s' % fname for fname in phone_fields),
+                'where_query': ' OR '.join('target.%s IS NOT NULL' % fname for fname in phone_fields),
+                'target_table': target._table
+            }
+            params = {'mailing_id': self.id, 'target_model': self.mailing_model_real}
+        else:
+            phone_fields = ['mobile', 'phone']
+            query = """
+                SELECT %(select_query)s
+                  FROM mailing_trace trace
+                  JOIN %(target_table)s target ON (trace.res_id = target.id)
+                  JOIN res_partner partner ON (target.partner_id = partner.id)
```

Comment:
Here you assume that the joining field is named "partner_id", but isn't that depending on the result of the "_sms_get_partner_fields" method?

I see that it can also be partner_idS apparently? Which changes the way you're joining as well.
To double-check.

Reply:
I limit the support to a m2o field explicitely now. Currently we have no sms marketing enabled model using a m2m to partners, so I don't try to support it as we do not support it on mail marketing either.